engines/native/sandbox.go

```diff
@@ -47,11 +47,28 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		}
 	}
 
-	// Create temporary user account
-	user, err := system.CreateUser(homeFolder.Path(), b.engine.groups)
-	if err != nil {
-		homeFolder.Remove() // best-effort clean-up this is a fatal error
-		return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+	var user *system.User
+	var username string
+
+	if b.engine.config.CreateUser {
+		// Create temporary user account
+		user, err = system.CreateUser(homeFolder.Path(), b.engine.groups)
+		if err != nil {
+			homeFolder.Remove() // best-effort clean-up this is a fatal error
+			return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+		}
+		username = user.Name()
+	} else {
```

Comment:
If we're introducing `system.CurrentUser()` I propose that we allow for that to be passed as `Owner` property, and instead do:
```go
var user *system.User
if b.engine.config.CreateUser {
  user, err = system.CreateUser(homeFolder.Path(), b.engine.groups)
  ...
} else {
  user, err = system.CurrentUser()
  ...
}

....
env["USER"] = user.Username()
```